include/boost/math/tools/polynomial.hpp

```diff
@@ -134,7 +236,11 @@ class polynomial
 
    // access:
    size_type size()const { return m_data.size(); }
-   size_type degree()const { return m_data.size() - 1; }
+   size_type degree()const
+   { 
+       BOOST_ASSERT(size() != 0);
```

Comment:
I also realize that I'm being slightly hypocritical, because I throw a `domain_error` in `quotient_remainder` for divide by zero. The same harsh logic above should apply here, but somehow it seemed OK to be kind in that case. Ultimately, I would stick to my guns of "user beware" in favour of efficiency and optionally have a less efficient interface that throws an exception.